core/src/main/java/com/netflix/conductor/core/config/Configuration.java

```diff
@@ -1,121 +1,137 @@
-	 * 
-	 */
-	public boolean disableAsyncWorkers();
-	
-	/**
-	 * 
-	 * @return ID of the server.  Can be host name, IP address or any other meaningful identifier.  Used for logging
-	 */
-	public String getServerId();
-
-	/**
-	 * 
-	 * @return Current environment. e.g. test, prod
-	 */
-	public String getEnvironment();
-
-	/**
-	 * 
-	 * @return name of the stack under which the app is running.  e.g. devint, testintg, staging, prod etc. 
-	 */
-	public String getStack();
-
-	/**
-	 * 
-	 * @return APP ID.  Used for logging
-	 */
-	public String getAppId();
-
-	/**
-	 * 
-	 * @return Data center region.  if hosting on Amazon the value is something like us-east-1, us-west-2 etc.
-	 */
-	public String getRegion();
-
-	/**
-	 * 
-	 * @return Availability zone / rack.  for AWS deployments, the value is something like us-east-1a, etc.
-	 */
-	public String getAvailabilityZone();
-
-	/**
-	 * 
-	 * @param name Name of the property
-	 * @param defaultValue  Default value when not specified
-	 * @return User defined integer property. 
-	 */
-	public int getIntProperty(String name, int defaultValue);
-	
-	/**
-	 * 
-	 * @param name Name of the property
-	 * @param defaultValue  Default value when not specified
-	 * @return User defined string property. 
-	 */
-	public String getProperty(String name, String defaultValue);
-	
-	
-	/**
-	 * 
-	 * @return Returns all the configurations in a map.
-	 */
-	public Map<String, Object> getAll();
-	
-	/**
-	 * 
-	 * @return Provides a list of additional modules to configure. 
-	 * Use this to inject additional modules that should be loaded as part of the Conductor server initialization
-	 * If you are creating custom tasks (com.netflix.conductor.core.execution.tasks.WorkflowSystemTask) then initialize them as part of the custom modules.
-	 */
-	public default List<AbstractModule> getAdditionalModules() {
-		return null;
-	}
+    String SWEEP_FREQUENCY_PROPERTY_NAME = "decider.sweep.frequency.seconds";
+    int SWEEP_FREQUENCY_DEFAULT_VALUE = 30;
+    String SWEEP_DISABLE_PROPERTY_NAME = "decider.sweep.disable";
+    // FIXME This really should be typed correctly.
+    String SWEEP_DISABLE_DEFAULT_VALUE = "false";
+    String DISABLE_ASYNC_WORKERS_PROPERTY_NAME = "conductor.disable.async.workers";
+    // FIXME This really should be typed correctly.
+    String DISABLE_ASYNC_WORKERS_DEFAULT_VALUE = "false";
+    String ENVIRONMENT_PROPERTY_NAME = "environment";
+    String ENVIRONMENT_DEFAULT_VALUE = "test";
+    String STACK_PROPERTY_NAME = "STACK";
+    String STACK_DEFAULT_VALUE = "test";
+    String APP_ID_PROPERTY_NAME = "APP_ID";
+    String APP_ID_DEFAULT_VALUE = "conductor";
+    String REGION_PROPERTY_NAME = "EC2_REGION";
+    String REGION_DEFAULT_VALUE = "us-east-1";
+    String AVAILABILITY_ZONE_PROPERTY_NAME = "EC2_AVAILABILITY_ZONE";
+    String AVAILABILITY_ZONE_DEFAULT_VALUE = "us-east-1c";
+    String ADDITIONAL_MODULES_PROPERTY_NAME = "conductor.additional.modules";
```

Comment:
If all of these above are meant as constants, could you mark them as `static final`? Also, is the intention to have default scope on these? Would a `private` or `public` visibility make more sense?

Reply:
Thanks for clarifying this @gorzell. Must admit that my Java chops are a little rusty. It's been a while since my Java 6 days. 🙇